internal/server/middleware/grpc/middleware.go

```diff
@@ -485,43 +494,25 @@ func AuditUnaryInterceptor(logger *zap.Logger, eventPairChecker EventPairChecker
 			return resp, err
 		}
 
-		action := audit.GRPCMethodToAction(info.FullMethod)
-
 		switch r := resp.(type) {
 		case *flipt.Flag:
-			if action != "" {
-				event = audit.NewEvent(audit.FlagType, action, actor, audit.NewFlag(r))
-			}
+			event = audit.NewEvent(request, actor, audit.NewFlag(r))
```

Comment:
we can likely clean this up a bit by having `audit.NewEvent` also know how to wrap the resp `r` with the appropriate type maybe? so we don't have to have this switch statement